javatests/arcs/core/entity/HandleManagerTestBase.kt

```diff
@@ -163,106 +165,120 @@ open class HandleManagerTestBase {
+        val writeHandle = writeHandleManager.createHandle(
+            HandleSpec(
+                "writeOnlySingleton",
+                HandleMode.Write,
+                SingletonType(EntityType(Person.SCHEMA)),
+                Person
+            ),
+            singletonKey
+        ).awaitReady() as WriteSingletonHandle<Person>
 
-        handleBUpdated = handleB.onUpdateDeferred {
-            log("handleB updated: $it")
-            it == null
-        }
-        handleA.dispatchClear()
-        handleBUpdated.await()
+        val readHandle = readHandleManager.createHandle(
+            HandleSpec(
+                "readOnlySingleton",
+                HandleMode.Read,
+                SingletonType(EntityType(Person.SCHEMA)),
+                Person
+            ),
+            singletonKey
+        ).awaitReady() as ReadSingletonHandle<Person>
 
-        assertThat(handleB.dispatchFetch()).isNull()
-    }
+        var received = CompletableDeferred<Person?>()
+        readHandle.onUpdate { received.complete(it) }
 
-    @Test
-    fun singleton_clearOnAClearDataWrittenByB() = testRunner {
-        val handleA = writeHandleManager.createSingletonHandle()
-        val handleB = readHandleManager.createSingletonHandle()
-        val handleBUpdated = handleB.onUpdateDeferred { it != null }
-        handleA.dispatchStore(entity1)
-        handleBUpdated.await()
+        // Verify store against empty.
+        writeHandle.dispatchStore(entity1)
+        assertThat(received.await()).isEqualTo(entity1)
+        assertThat(readHandle.dispatchFetch()).isEqualTo(entity1)
 
-        // Now read back from a different handle
-        val updateADeferred = handleA.onUpdateDeferred { it == null }
-        handleB.dispatchClear()
-        assertThat(handleB.dispatchFetch()).isNull()
+        // Verify store overwrites existing.
+        received = CompletableDeferred()
+        writeHandle.dispatchStore(entity2)
+        assertThat(received.await()).isEqualTo(entity2)
+        assertThat(readHandle.dispatchFetch()).isEqualTo(entity2)
 
-        updateADeferred.await()
-        assertThat(handleA.dispatchFetch()).isNull()
+        // Verify clear.
+        received = CompletableDeferred()
+        writeHandle.dispatchClear()
+        assertThat(received.await()).isNull()
+        assertThat(readHandle.dispatchFetch()).isNull()
     }
 
     @Test
-    fun singleton_writeAndOnUpdate() = testRunner {
-        val writeHandle = writeHandleManager.createSingletonHandle()
-            as WriteSingletonHandle<Person>
-
-        // Now read back from a different handle
-        val readHandle = readHandleManager.createSingletonHandle()
+    fun singleton_writeAndReadBack_bidirectional() = testRunner {
+        // Read/write handle <-> read/write handle
+        val handle1 = writeHandleManager.createHandle(
+            HandleSpec(
+                "readWriteSingleton1",
+                HandleMode.ReadWrite,
```

Comment:
if you use ReadWrite, isn't this the same as createSingletonHandle?

Reply:
This is for consistency with the other tests using a similar pattern (and also because I think having the details explicit directly in the test is more useful than saving a bit of redundant code). Note that the collection version of the same test *isn't* the same as `createCollectionHandle` - I don't want the Query perm set.